api/app/signals/apps/api/v1/views/pdf.py

```diff
@@ -74,6 +81,49 @@ def get_object(self):
         self.check_object_permissions(request=self.request, obj=obj)
         return obj
 
+    def _resize(self, image):
+        # Consider image orientation:
+        if image.width > image.height:
+            # landscape
+            width = self.max_size
+            height = int((self.max_size / image.width) * image.height)
+        else:
+            # portrait
+            width = int((self.max_size / image.height) * image.width)
+            height = self.max_size
+
+        return image.resize(size=(width, height), resample=Image.LANCZOS).convert('RGB')
+
+    def _get_context_data_images(self, signal):
+        jpg_data_urls = []
+        for att in signal.attachments.all():
+            # Attachment is_image property is currently not reliable
+            _, ext = os.path.splitext(att.file.name)
+            if ext not in ['.gif', '.jpg', '.jpeg', '.png']:
+                continue  # unsupported image format, or not image format
+
+            try:
+                with default_storage.open(att.file.name) as file:
+                    buffer = io.BytesIO(file.read())
+                image = Image.open(buffer)
+            except (UnidentifiedImageError, FileNotFoundError):
+                continue  # Protect against missing files and bad data.
+            except:  # noqa:E722
+                msg = f'Cannot open image attachment pk={att.pk}'
+                logger.warn(msg, exc_info=True)
+                continue
+
+            if image.width > self.max_size or image.height > self.max_size:
+                image = self._resize(image)
+
+            new_buffer = io.BytesIO()
+            image.save(new_buffer, format='JPEG')
+            encoded = 'data:image/jpg;base64,' + base64.b64encode(new_buffer.getvalue()).decode('utf-8')
```

Comment:
I like f strings more than concatenating strings with a `+`

Reply:
Generally agree, but I do not think that
`f'data:image/jpg;base64,{base64.b64encode(new_buffer.getvalue()).decode('utf-8')}' is shorter or cleaner.

Reply:
Will consider